Compilation error

During a recent migration on Bioconda we ran into the following error while rebuilding aodp:

```
2022-10-04T09:43:59.5065431Z 09:43:59 BIOCONDA INFO (OUT) $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-c++ -c -fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/aodp-2.5.0.1 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -Wall -o o/Source.o c/Source.cpp -Ih -Ily
2022-10-04T09:43:59.8140776Z 09:43:59 BIOCONDA INFO (OUT) In file included from /opt/conda/conda-bld/aodp_1664876457391/_build_env/x86_64-conda-linux-gnu/include/c++/10.4.0/map:60,
2022-10-04T09:43:59.8164258Z 09:43:59 BIOCONDA INFO (OUT)                  from h/TrieSlice.h:11,
2022-10-04T09:43:59.8165466Z 09:43:59 BIOCONDA INFO (OUT)                  from h/Match.h:13,
2022-10-04T09:43:59.8166488Z 09:43:59 BIOCONDA INFO (OUT)                  from c/Match.cpp:3:
2022-10-04T09:43:59.8170434Z 09:43:59 BIOCONDA INFO (OUT) /opt/conda/conda-bld/aodp_1664876457391/_build_env/x86_64-conda-linux-gnu/include/c++/10.4.0/bits/stl_tree.h: In instantiation of 'static const _Key& std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_S_key(std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_Const_Link_type) [with _Key = unsigned int; _Val = unsigned int; _KeyOfValue = std::_Identity<unsigned int>; _Compare = Match::onSequence(const string&, const Cover<unsigned int>&, Alignment&)::ClusterSizeCompare; _Alloc = std::allocator<unsigned int>; std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_Const_Link_type = const std::_Rb_tree_node<unsigned int>*]':
2022-10-04T09:43:59.8176603Z 09:43:59 BIOCONDA INFO (OUT) /opt/conda/conda-bld/aodp_1664876457391/_build_env/x86_64-conda-linux-gnu/include/c++/10.4.0/bits/stl_tree.h:808:22:   required from 'static const _Key& std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_S_key(std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_Const_Base_ptr) [with _Key = unsigned int; _Val = unsigned int; _KeyOfValue = std::_Identity<unsigned int>; _Compare = Match::onSequence(const string&, const Cover<unsigned int>&, Alignment&)::ClusterSizeCompare; _Alloc = std::allocator<unsigned int>; std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_Const_Base_ptr = const std::_Rb_tree_node_base*]'
2022-10-04T09:43:59.8181478Z 09:43:59 BIOCONDA INFO (OUT) /opt/conda/conda-bld/aodp_1664876457391/_build_env/x86_64-conda-linux-gnu/include/c++/10.4.0/bits/stl_tree.h:2209:40:   required from 'std::pair<std::_Rb_tree_node_base*, std::_Rb_tree_node_base*> std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_M_get_insert_hint_unique_pos(std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::const_iterator, const key_type&) [with _Key = unsigned int; _Val = unsigned int; _KeyOfValue = std::_Identity<unsigned int>; _Compare = Match::onSequence(const string&, const Cover<unsigned int>&, Alignment&)::ClusterSizeCompare; _Alloc = std::allocator<unsigned int>; std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::const_iterator = std::_Rb_tree<unsigned int, unsigned int, std::_Identity<unsigned int>, Match::onSequence(const string&, const Cover<unsigned int>&, Alignment&)::ClusterSizeCompare, std::allocator<unsigned int> >::const_iterator; std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::key_type = unsigned int]'
2022-10-04T09:43:59.8190980Z 09:43:59 BIOCONDA INFO (OUT) /opt/conda/conda-bld/aodp_1664876457391/_build_env/x86_64-conda-linux-gnu/include/c++/10.4.0/bits/stl_tree.h:2269:4:   required from 'std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::iterator std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_M_insert_unique_(std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::const_iterator, _Arg&&, _NodeGen&) [with _Arg = const unsigned int&; _NodeGen = std::_Rb_tree<unsigned int, unsigned int, std::_Identity<unsigned int>, Match::onSequence(const string&, const Cover<unsigned int>&, Alignment&)::ClusterSizeCompare, std::allocator<unsigned int> >::_Alloc_node; _Key = unsigned int; _Val = unsigned int; _KeyOfValue = std::_Identity<unsigned int>; _Compare = Match::onSequence(const string&, const Cover<unsigned int>&, Alignment&)::ClusterSizeCompare; _Alloc = std::allocator<unsigned int>; std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::iterator = std::_Rb_tree<unsigned int, unsigned int, std::_Identity<unsigned int>, Match::onSequence(const string&, const Cover<unsigned int>&, Alignment&)::ClusterSizeCompare, std::allocator<unsigned int> >::iterator; std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::const_iterator = std::_Rb_tree<unsigned int, unsigned int, std::_Identity<unsigned int>, Match::onSequence(const string&, const Cover<unsigned int>&, Alignment&)::ClusterSizeCompare, std::allocator<unsigned int> >::const_iterator]'
2022-10-04T09:43:59.8203234Z 09:43:59 BIOCONDA INFO (OUT) /opt/conda/conda-bld/aodp_1664876457391/_build_env/x86_64-conda-linux-gnu/include/c++/10.4.0/bits/stl_tree.h:1111:23:   required from 'std::__enable_if_t<std::is_same<_Val, typename std::iterator_traits<_InputIterator>::value_type>::value> std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_M_insert_range_unique(_InputIterator, _InputIterator) [with _InputIterator = std::_Rb_tree_const_iterator<unsigned int>; _Key = unsigned int; _Val = unsigned int; _KeyOfValue = std::_Identity<unsigned int>; _Compare = Match::onSequence(const string&, const Cover<unsigned int>&, Alignment&)::ClusterSizeCompare; _Alloc = std::allocator<unsigned int>; std::__enable_if_t<std::is_same<_Val, typename std::iterator_traits<_InputIterator>::value_type>::value> = void; typename std::iterator_traits<_InputIterator>::value_type = unsigned int]'
2022-10-04T09:43:59.8205727Z 09:43:59 BIOCONDA INFO (OUT) /opt/conda/conda-bld/aodp_1664876457391/_build_env/x86_64-conda-linux-gnu/include/c++/10.4.0/bits/stl_set.h:212:31:   required from 'std::set<_Key, _Compare, _Alloc>::set(_InputIterator, _InputIterator, const _Compare&, const allocator_type&) [with _InputIterator = std::_Rb_tree_const_iterator<unsigned int>; _Key = unsigned int; _Compare = Match::onSequence(const string&, const Cover<unsigned int>&, Alignment&)::ClusterSizeCompare; _Alloc = std::allocator<unsigned int>; std::set<_Key, _Compare, _Alloc>::allocator_type = std::allocator<unsigned int>]'
2022-10-04T09:43:59.8206656Z 09:43:59 BIOCONDA INFO (OUT) c/Match.cpp:154:112:   required from here
2022-10-04T09:43:59.8207419Z 09:43:59 BIOCONDA INFO (OUT) /opt/conda/conda-bld/aodp_1664876457391/_build_env/x86_64-conda-linux-gnu/include/c++/10.4.0/bits/stl_tree.h:782:8: error: static assertion failed: comparison object must be invocable as const
2022-10-04T09:43:59.8208107Z 09:43:59 BIOCONDA INFO (OUT)   782 |        is_invocable_v<const _Compare&, const _Key&, const _Key&>,
2022-10-04T09:43:59.8208484Z 09:43:59 BIOCONDA INFO (OUT)       |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2022-10-04T09:44:00.0203043Z 09:44:00 BIOCONDA INFO (OUT) make: *** [Makefile:129: o/Match.o] Error 1
```

Please advise.

--- a/.github/workflows/cpp-build.yml
+++ b/.github/workflows/cpp-build.yml
@@ -28,7 +28,7 @@ jobs:
         conda info
         conda list
     - name: Install BioPerl >=1.7.2
-      run: conda install "perl-bioperl>=1.7.2" "gxx_linux-64 ==7.3.0"
+      run: conda install "perl-bioperl>=1.7.2" "gxx_linux-64 ==7.3.0" "perl-list-moreutils"
     - name: g++ version
       run: g++ --version
     - name: configure

--- a/c/Application.cpp
+++ b/c/Application.cpp
@@ -6,7 +6,7 @@
 // overwrites stuff between "" in the line below magic
 // see the Makefile in the project root
 // magic (do not modify): iMMmwyWGTTGeJ6TYQg2myA
-const string Application::version = "2.5.0.1"; // do not modify
+const string Application::version = "2.5.0.2"; // do not modify
 
 const int Application :: I_oligo_size_min = 12;
 const int Application :: I_oligo_size_max = 100;

--- a/c/Match.cpp
+++ b/c/Match.cpp
@@ -137,7 +137,7 @@ void Match::onSequence( const string& na, const Cover<Position>& amb, Alignment&
 
 // 	(2) build the minimum set of sequences
 	struct ClusterSizeCompare{
-		bool operator() ( const Cluster& c1, const Cluster& c2 ) {
+		bool operator() ( const Cluster& c1, const Cluster& c2 ) const {
 			const size_t s1 = clusters.at( c1 ).size();
 			const size_t s2 = clusters.at( c2 ).size();
 

--- a/c/aodp.help.cpp
+++ b/c/aodp.help.cpp
@@ -2,7 +2,7 @@
 
 const string Application::_help =
 "NAME\n"
-"     aodp - Automated Oligonucleotide Design Pipeline version \"2.5.0.1\"\n"
+"     aodp - Automated Oligonucleotide Design Pipeline version \"2.5.0.2\"\n"
 "\n"
 "SYNOPSIS\n"
 "    aodp [*options*] *output* *fasta-sequence-file...*\n"

--- a/man/Makefile
+++ b/man/Makefile
@@ -8,7 +8,7 @@
 # see the Makefile in the project root
 # magic line below (do not modify)
 # iMMmwyWGTTGeJ6TYQg2myA
-VERSION= "2.5.0.1"
+VERSION= "2.5.0.2"
 
 PERL_DIR= ../pl
 POD_DIR= pod

--- a/man/pod/aodp
+++ b/man/pod/aodp
@@ -9,7 +9,7 @@ see the Makefile in the project root
 magic line below (do not modify)
 
 L<|iMMmwyWGTTGeJ6TYQg2myA>
-aodp - Automated Oligonucleotide Design Pipeline version "2.5.0.1"
+aodp - Automated Oligonucleotide Design Pipeline version "2.5.0.2"
 
 =head1 SYNOPSIS
 

--- a/pl/clado
+++ b/pl/clado
@@ -54,7 +54,7 @@ see the Makefile in the project root
 magic line below (do not modify)
 
 L<|iMMmwyWGTTGeJ6TYQg2myA>
-Cladogram generator (clado) version "2.5.0.1"
+Cladogram generator (clado) version "2.5.0.2"
 
 =head1 SYNOPSIS
 

--- a/pl/clus
+++ b/pl/clus
@@ -292,7 +292,7 @@ see the Makefile in the project root
 magic line below (do not modify)
 
 L<|iMMmwyWGTTGeJ6TYQg2myA>
-Cluster oligo signature analyzer (C<clus>) version "2.5.0.1"
+Cluster oligo signature analyzer (C<clus>) version "2.5.0.2"
 
 =head1 SYNOPSIS
 

--- a/pl/tax2nwk
+++ b/pl/tax2nwk
@@ -89,7 +89,7 @@ see the Makefile in the project root
 magic line below (do not modify)
 
 L<|iMMmwyWGTTGeJ6TYQg2myA>
-Taxonomy to Newick convertor (C<tax2nwk>) version "2.5.0.1"
+Taxonomy to Newick convertor (C<tax2nwk>) version "2.5.0.2"
 
 =head1 SYNOPSIS
 

